content/local/writer.go

```diff
@@ -58,18 +60,24 @@ func (w *writer) Status() (content.Status, error) {
 //
 // Cannot be called concurrently with `Write`.
 func (w *writer) Digest() digest.Digest {
-	return w.digester.Digest()
+	for {
+		if atomic.LoadUint64(&w.inUse) == 0 {
+			return w.digester.Digest()
+		}
+	}
 }
 
 // Write p to the transaction.
 //
 // Note that writes are unbuffered to the backing file. When writing, it is
 // recommended to wrap in a bufio.Writer or, preferably, use io.CopyBuffer.
 func (w *writer) Write(p []byte) (n int, err error) {
+	atomic.StoreUint64(&w.inUse, 1)
```

Comment:
Why use `atomic` instead of `sync.Mutex`?  Note that the use of `StoreUint64` here doesn't actually prevent `Write` from being called concurrently; you'd need `CompareAndSwapUint64` instead, or `sync.Mutex.Lock()` for a higher-level mutex with correct behavior.

Reply:
Sure. sync.Mutex looks more promising. Thanks